rest-api/src/main/java/gov/cms/qpp/conversion/api/controllers/v1/QrdaControllerV1.java

```diff
@@ -68,16 +69,29 @@ public QrdaControllerV1(final QrdaService qrdaService, final ValidationService v
 
 		validationService.validateQpp(conversionReport);
 
-		auditService.success(conversionReport);
+		Metadata metadata = audit(conversionReport);
 
 		API_LOG.info("Conversion request succeeded");
 
 		HttpHeaders httpHeaders = new HttpHeaders();
 		httpHeaders.setContentType(MediaType.APPLICATION_JSON_UTF8);
 
+		if (metadata != null) {
+			httpHeaders.add("Location", metadata.getUuid());
+		}
+
 		return new ResponseEntity<>(conversionReport.getEncoded().toString(), httpHeaders, HttpStatus.CREATED);
 	}
 
+	private Metadata audit(Converter.ConversionReport conversionReport) {
+		try {
+			CompletableFuture<Metadata> metadata = auditService.success(conversionReport);
+			return metadata == null ? null : metadata.get();
+		} catch (InterruptedException | ExecutionException exception) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)

Reply:
As written, this isn't clear as to why we're ignoring the exceptions.  If there's a reason, please comment here, otherwise we should do something with the exceptions.

Reply:
Agreed